File: 1. twoSum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        result = []
        
        print(nums)
        for i in range(len(nums)):
            match = -1
            remain = target - nums[i]
            print("i: ", i)
            print("remain: ", remain)
            for c in range(i+1, len(nums)):
                print(nums[c])
                if( remain - nums[c] == 0 ):
                    print("c: ", c)
                    match = c
                    break
            if(match is not -1):
                result.append(i)
                result.append(match)
                break

        return result
        

c = Solution()
x = c.twoSum([-1,-2,-3,-4,-5], -8)
#x = c.twoSum([0,4,3,0], 0)
#x = c.twoSum([2,7,11,13], 9)
print(x)

File: 1. twoSum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    //判斷每陣列中一個元素是否比target小
    let result = Array();
    for(let i = 0; i<nums.length; i++){
        result = [];
        let match = -1;
        let remain = target - nums[i];
        for(let j = i+1; j < nums.length; j++){
            if(remain - nums[j] == 0){
                match = j
                break
            }
        }
        if( match != -1 ){
            result.push(i)
            result.push(match)
            break
        }
    }
    
    return result
};


let nums = [2, 7, 11, 13];
let target = 9;

var result = twoSum( nums, target );
console.log(result)